### ISSUE TITLE
BUG: np.mean does not raise AxisError for illegal axes

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#15817]
Minor, but:
```python
>>> a = np.arange(10)
>>> a.sum(axis=2)
AxisError: axis 2 is out of bounds for array of dimension 1

>>> a.mean(axis=2)
# from _count_reduce_items in methods.py
IndexError: tuple index out of range
```

I'd expect both to raise the same error

**Comment by sgasse** [comment]
Hey @eric-wieser, I created a pull request fixing this bug. It looks like `_count_reduce_items` is also used in `np.var`, so I added tests for both `np.mean` and `np.var`. Would be great if you could review my PR!

**Comment by eric-wieser** [comment]
Closed in #15836
